### PR TITLE
Add 'client only' build option

### PR DESCRIPTION
**Pull request by pasnox**
Task-Id: KDEND-225

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -170,14 +174,30 @@ set(GAMMARAY_BUILD_UI_DEFAULT ON)
...
 
+gammaray_option(
+    GAMMARAY_CLIENT_ONLY_BUILD
+    "Build the client part only."
+    OFF
+)
+
+if (GAMMARAY_CLIENT_ONLY_BUILD)
+    set(GAMMARAY_BUILD_UI_DEFAULT ON)
+    set(BUILD_TESTING OFF)
```
Comment (vkrause) [inline CMakeLists.txt]: Why disable testing in this case?

File: CMakeLists.txt [hunk 2]
```diff
@@ -752,10 +772,14 @@ if(GAMMARAY_BUILD_UI AND (Qt5Widgets_FOUND OR QT_QTGUI_FOUND))
...
   endif()
   add_subdirectory(ui)
-  add_subdirectory(inprocessui)
+  if (NOT GAMMARAY_CLIENT_ONLY_BUILD)
+    add_subdirectory(inprocessui)
+  endif()
   if(NOT GAMMARAY_PROBE_ONLY_BUILD)
     add_subdirectory(client)
-    add_subdirectory(app)
+    if (NOT GAMMARAY_CLIENT_ONLY_BUILD)
```
Comment (vkrause) [inline CMakeLists.txt]: Isn't "app" the macos client/launcher combination and thus should be build in client-only mode too?

Comment (pasnox) [reply CMakeLists.txt]: Fixed

File: CMakeLists.txt (hunk 1 above)
Comment (pasnox) [reply CMakeLists.txt]: Currently they caused lots of build issues in those, I will have a closer looks now that there is no much comments for the review.

Comment (pasnox) [reply CMakeLists.txt]: Fixed

**Pull request merged**